### PR TITLE
Update pre-commit to run black 22.3.0

### PR DESCRIPTION
**Pull request by vyasr**
This PR updates us to use black 22.3.0, which is now necessary because older versions of black are not compatible with current versions of Click (see https://github.com/psf/black/issues/2964 is resolved). 

I've opened this for 22.06 since [I don't see any open PRs attempting to merge into 22.04](https://github.com/rapidsai/cudf/pulls?q=is%3Apr+is%3Aopen+base%3Abranch-22.04) anymore, but this issue will block CI (which runs style checks using pre-commit) so if necessary I can backport.

### CODE REVIEW

**Review by bdice — APPROVED** [review]
I am seeing CI failures. This should fix the problem until we can identify a longer-term fix (which may require upgrading black).

**Review by galipremsagar — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Comment by vyasr** [comment]
Just FYI I'm following up on what to do regarding the outstanding copyright changes here,

File: python/cudf/cudf/_lib/column.pyi [hunk 1]
```diff
@@ -32,94 +32,46 @@ class Column:
...
+    def children(self) -> Tuple[ColumnBase, ...]: ...
+    def set_base_children(self, value: Tuple[ColumnBase, ...]) -> None: ...
+    def _mimic_inplace(
+        self, other_col: ColumnBase, inplace=False
+    ) -> Optional[ColumnBase]: ...
     @staticmethod
-    def from_scalar(
-        val: ScalarLike,
-        size: int
-    ) -> ColumnBase:  # TODO: This should be Scalar, not ScalarLike
```
Comment (bdice) [inline python/cudf/cudf/_lib/column.pyi:124]: @vyasr FYI this reformat drops comments in `.pyi` files. I noticed this in a previous attempt to update `black` but I wasn't sure how to address it.

Comment (vyasr) [reply python/cudf/cudf/_lib/column.pyi:124]: Thanks for the note. I'll see what we can do about that.

Comment (vyasr) [reply python/cudf/cudf/_lib/column.pyi:124]: It looks like the issue is specifically the inline comments on lines that get compressed. I guess we'll just have to remember to put comments on separate lines from now on in pyi files.

Comment (jakirkham) [reply python/cudf/cudf/_lib/column.pyi:124]: With `TODO`s, wonder if we should just file these as issues. That would make them easier to track, cross-reference, triage, resolve, etc.

Comment (jakirkham) [reply python/cudf/cudf/_lib/column.pyi:124]: > ...at the moment it's not uncommon for TODOs like this to become moot when the relevant code paths are removed wholesale

Honestly this is my biggest worry with `TODO`s and have seen this happen to other code bases. Comments left that no longer pertain to the code they are near with no one that has a clue what they mean. It is clearer to see what issues intended and when they are resolved.

**Pull request merged**